dipy/segment/tests/test_metric.py

```diff
@@ -174,7 +174,7 @@ def dist(self, v1, v2):
             if metric.are_compatible(f1.shape, f2.shape):
                 distance = metric.dist(f1, f2)
                 if np.all(f1 == f2):
-                    assert_equal(distance, 0.)
+                    assert_almost_equal(distance, 0.)
```

Comment:
I agree with you, I thought casting them in `float64` would have avoided this precision loss.
Wait! I'm stupid :P the error is probably not with the Python version but rather in the Cython one!

In the following, both `features1` and `features2` are C float whereas `cos_theta` is a C double. If I'm not mistaken, the multiplication would be perform in 32 bits instead of 64, right?
```python
cos_theta += features1[0, d] * features2[0, d]
```